src/btree/bt_handle.c

```diff
@@ -760,6 +760,62 @@ __btree_tree_open_empty(WT_SESSION_IMPL *session, bool creation)
     return (ret);
 }
 
+/*
+ * __wt_btree_new_internal_page --
+ *     Create an empty internal page, as well as a leaf page that is a child of the internal page.
```

Comment:
I think this probably should just create a new empty internal page - that page will be traversed by the walk, and traversing into the page will cause the child to be created. I suggest making that change, before digging into the IKEY problem further.